pintc/src/asm_gen/asm_builder.rs

```diff
@@ -1090,11 +1090,22 @@ impl<'a> AsmBuilder<'a> {
             ));
         }
 
+        // check it's a prime op
+        let tuple_expr = if let Expr::UnaryOp {
+            op: UnaryOp::NextState,
+            expr,
+            ..
+        } = tuple.get(contract)
+        {
+            expr
+        } else {
+            tuple
+        };
```

Comment:
It feels like this is too specific and we should have a more general solution for getting the tuple expression, but I guess as this stage any other 'wrapper' has been lowered so it's OK.

OTOH, perhaps we should be pre-resolving state accesses as pre- or post- and then removing the next-state ops altogether.

So this is fine for now, but I think it might not be a long term solution.